encfs/CipherFileIO.cpp

```diff
@@ -168,18 +168,33 @@ int CipherFileIO::getAttr(struct stat *stbuf) const {
   // stat() the backing file
   int res = base->getAttr(stbuf);
 
-  // adjust size if we have a file header
-  if ((res == 0) && haveHeader && S_ISREG(stbuf->st_mode) &&
-      (stbuf->st_size > 0)) {
+  // adjust size if we have a file header or padding
+  if ((res == 0) && S_ISREG(stbuf->st_mode)) {
     if (!fsConfig->reverseEncryption) {
       /* In normal mode, the upper file (plaintext) is smaller
        * than the backing ciphertext file */
-      rAssert(stbuf->st_size >= HEADER_SIZE);
-      stbuf->st_size -= HEADER_SIZE;
-    } else {
+      if (haveHeader && (stbuf->st_size > 0)) {
+      	/* Instead of using rAssert, we could also relax the rule and return 0.
+      	 * Think about a file which would have been partially written.
+      	 * Same for getSize() below. */
+        rAssert(stbuf->st_size >= HEADER_SIZE);
+        stbuf->st_size -= HEADER_SIZE;
+      }
+      if (haveCBCPadding && (stbuf->st_size > 0)) {
+        rAssert(stbuf->st_size >= fsConfig->cipher->cipherBlockSize());
+        stbuf->st_size -= fsConfig->cipher->cipherBlockSize();
+        stbuf->st_size -= stbuf->st_size / (blockSize() + 1);
```

Comment:
This is rounding of some kind? Needs a comment, or better, it's own function. In gocryptfs I have `cipherSizeToPlainSize()` and `plainSizeToCipherSize()`.